doc/src/manual/performance-tips.md

```diff
@@ -577,6 +577,45 @@ c = (b + 1.0f0)::Complex{T}
 does not hinder performance (but does not help either) since the compiler can determine the type of `c`
 at the time `k` is compiled.
 
+### Beware of suboptimal type assertion patterns
+
+Due to [limitations](https://github.com/JuliaLang/julia/issues/38274) in current versions
+of Julia, some [type assertion](@ref "Type Declarations") patterns do not help refine type
+inference, even though, intuitively, they should. An example:
```

Comment:
```suggestion
inference. An example:
```

IMO documentation should consist of positive statements and omit normative statements.